packages/studio-ui/src/components/UndoRedo.tsx

```diff
@@ -22,6 +22,20 @@ export default function UndoRedo(): JSX.Element {
     redo();
   }, [redo]);
 
+  const handleUndoKeydown = useCallback(
+    (event: KeyboardEvent) => {
+      event.preventDefault();
+      if ((event.ctrlKey || event.metaKey) && event.key === "z") {
+        undo();
+      }
+    },
+    [undo]
+  );
+
+  useEffect(() => {
+    document.addEventListener("keydown", handleUndoKeydown);
```

Comment:
Note that since the reference to `undo` does not change, this `useEffect` will only be run on first render.  Furthermore (fun fact), if you call `addEventListener` with a named function, it will not be added more than once.  So, if there was a case that the `useEffect` hook would be called twice, it would not add a duplicate listener.